keps/sig-node/4381-dra-numeric-parameters/README.md

```diff
@@ -0,0 +1,1080 @@
+
+- [ ] (R) Enhancement issue in release milestone, which links to KEP dir in [kubernetes/enhancements] (not the initial KEP PR)
+- [ ] (R) KEP approvers have approved the KEP status as `implementable`
+- [ ] (R) Design details are appropriately documented
+- [ ] (R) Test plan is in place, giving consideration to SIG Architecture and SIG Testing input (including test refactors)
+  - [ ] e2e Tests for all Beta API Operations (endpoints)
+  - [ ] (R) Ensure GA e2e tests meet requirements for [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+  - [ ] (R) Minimum Two Week Window for GA e2e tests to prove flake free
+- [ ] (R) Graduation criteria is in place
+  - [ ] (R) [all GA Endpoints](https://github.com/kubernetes/community/pull/1806) must be hit by [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+- [ ] (R) Production readiness review completed
+- [ ] (R) Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+Dynamic Resource Allocation uses parameters for resources that are completely
+opaque to core Kubernetes. They get interpreted by the DRA driver's controller
+(for allocating claims) and the DRA drivers' kubelet plugin (for configuring
+the resource on a node). During scheduling of a pod, kube-scheduler and the DRA
+driver controller(s) that handle claims used by the pod communicate through the
+apiserver by updating the PodSchedulingContext object, ultimately leading to
+allocation of all pending claims and the pod being scheduled onto a node.
+
+This approach poses a problem for [Cluster
+Autoscaler](https://github.com/kubernetes/autoscaler) (CA) because it cannot
+simulate the effect of allocating or deallocating claims. "Numeric parameters"
+is an extension of DRA that addresses this by making claim parameters less
+opaque. "Numeric models" define how resource capacity per node can be published
+and the claim parameters that allocate some of that capacity for a claim.
+
+This KEP describes the core changes needed for this. Specific numeric models
+get defined in separate KEPs. Developers will be able to add their own numeric
```

Comment:
True. But people are using custom schedulers today, for whatever reason, in environments where they can be used. It would be good to support such scenarios.

Reply:
Custom schedulers are no longer supported, therefore I am closing this conversation.